pkg/applyconfigurations/testdata/cronjob_types.go

```diff
@@ -0,0 +1,309 @@
+	// +optional
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit,omitempty"`
+
+	// This tests byte slices are allowed as map values.
+	ByteSliceData map[string][]byte `json:"byteSliceData,omitempty"`
+
+	// This tests string slices are allowed as map values.
+	StringSliceData map[string][]string `json:"stringSliceData,omitempty"`
+
+	// This tests pointers are allowed as map values.
+	PtrData map[string]*string `json:"ptrData,omitempty"`
+
+	// This tests that markers that are allowed on both fields and types are applied to fields
+	// +kubebuilder:validation:MinLength=4
+	TwoOfAKindPart0 string `json:"twoOfAKindPart0"`
+
+	// This tests that markers that are allowed on both fields and types are applied to types
+	TwoOfAKindPart1 LongerString `json:"twoOfAKindPart1"`
+
+	// This tests that primitive defaulting can be performed.
+	// +kubebuilder:default=forty-two
+	DefaultedString string `json:"defaultedString"`
+
+	// This tests that slice defaulting can be performed.
+	// +kubebuilder:default={a,b}
+	DefaultedSlice []string `json:"defaultedSlice"`
+
+	// This tests that object defaulting can be performed.
+	// +kubebuilder:default={{nested: {foo: "baz", bar: true}},{nested: {bar: false}}}
+	DefaultedObject []RootObject `json:"defaultedObject"`
+
+	// This tests that pattern validator is properly applied.
+	// +kubebuilder:validation:Pattern=`^$|^((https):\/\/?)[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|\/?))$`
+	PatternObject string `json:"patternObject"`
+
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:validation:nullable
+	EmbeddedResource runtime.RawExtension `json:"embeddedResource"`
+
+	// +kubebuilder:validation:nullable
+	// +kubebuilder:pruning:PreserveUnknownFields
+	UnprunedJSON NestedObject `json:"unprunedJSON"`
+
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:validation:nullable
+	UnprunedEmbeddedResource runtime.RawExtension `json:"unprunedEmbeddedResource"`
+
+	// This tests that a type-level pruning maker works.
+	UnprunedFromType Preserved `json:"unprunedFomType"`
+
+	// This tests that associative lists work.
+	// +listType=map
+	// +listMapKey=name
+	// +listMapKey=secondary
+	AssociativeList []AssociativeType `json:"associativeList"`
+
+	// A map that allows different actors to manage different fields
+	// +mapType=granular
+	MapOfInfo map[string][]byte `json:"mapOfInfo"`
+
+	// A struct that can only be entirely replaced
+	// +structType=atomic
+	StructWithSeveralFields NestedObject `json:"structWithSeveralFields"`
+
+	// This tests that type references are properly flattened
+	// +kubebuilder:validation:optional
+	JustNestedObject *JustNestedObject `json:"justNestedObject,omitempty"`
+
+	// This tests that min/max properties work
+	MinMaxProperties MinMaxObject `json:"minMaxProperties,omitempty"`
+}
+
+// +kubebuilder:validation:Type=object
+// +kubebuilder:pruning:PreserveUnknownFields
+type Preserved struct {
```

Comment:
Could you say a bit more (maybe in godoc or a comment) about how this type works? I see that the marshaling has been overridden but it's not totally clear to me what this case exemplifies.